### PR TITLE
Update to .NET 8 and Remove .NET Core 3.1

### PR DESCRIPTION
**Pull request by aatchison14**
update target frameworks to remove .NET Core 3.1 and add .NET 8

### CODE REVIEW

File: Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs [hunk 1]
```diff
@@ -24,10 +24,10 @@ public static class ResponseWriterTests
         var httpContext = CreateHttpContext();
 
         // Act
-        await RockLibHealthChecks.ResponseWriter(httpContext, healthReport).ConfigureAwait(false);
+        await RockLibHealthChecks.ResponseWriter(httpContext, healthReport).ConfigureAwait(true);
```
Comment (JasonBock) [inline Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs:27]: Why are we switching these to `true`?

File: Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs [hunk 2]
```diff
@@ -145,29 +145,31 @@ public static class ResponseWriterTests
             .BeFalse();
     }
 
-    private static IEnumerable<JToken> GetHealthCheckEntries(string jsonBody) => 
-        JObject.Parse(jsonBody)!
+    private static IEnumerable<JToken> GetHealthCheckEntries(string jsonBody)
```
Comment (JasonBock) [inline Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs:148]: Was there a reason to change to not be an expression-bodied member?

File: Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs [hunk 3]
```diff
@@ -145,29 +145,31 @@ public static class ResponseWriterTests
...
     {
         httpContext.Response.Body.Seek(0, SeekOrigin.Begin);
 
-        string bodyStr;
-        using (var reader = new StreamReader(httpContext.Response.Body))
-        {
-            bodyStr = await reader.ReadToEndAsync().ConfigureAwait(false);
-        }
+        using var reader = new StreamReader(httpContext.Response.Body);
+        var bodyStr = await reader.ReadToEndAsync().ConfigureAwait(false);
```
Comment (JasonBock) [inline Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs:161]: Typically, in this case I wouldn't declare a variable and then return it. Just return the result of `ReadToEndAsync()` - same with what's done in `CreateHttpContext()`.

**Review by JasonBock — CHANGES_REQUESTED** [review]

File: Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs (hunk 3 above)
Comment (aatchison14) [reply Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs:161]: yeah, I completely missed that

File: Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs (hunk 2 above)
Comment (aatchison14) [reply Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs:148]: another analyzer suggestion: Code body does not conform to code style settings: use statement body. I'll put it back

Comment (JasonBock) [reply Tests/RockLib.HealthChecks.AspNetCore.ResponseWriter.Tests/ResponseWriterTests.cs]: I think we have an inconsistency across our .editorconfig settings with this one. Personally I like expression-bodied members :)

**Review by JasonBock — APPROVED** [review]

**Review by motionsuggests — APPROVED** [review]

**Review by NeilSchiefer — APPROVED** [review]

**Pull request merged**